guides-and-standards/guide-choosing-software-dependencies.md

```diff
@@ -0,0 +1,59 @@
+
+**When working on a software tool or workflow you anticipate that you or another Arcadian will use multiple times, you need to be careful about which tools you include**. This is _not_ the same as a workflow that documents the steps taken in an analysis project and is often bespoke and specific to a given task. Examples of existing tools or workflows where we should be careful about the quality of tools we use include:
+
+- nf-core-style Nextflow workflows like [NovelTree](https://github.com/Arcadia-Science/noveltree), [PreHGT](https://github.com/Arcadia-Science/prehgt), and [reads2genome](https://github.com/Arcadia-Science/reads2genome)
+- Snakemake workflows written as re-usable tools like [ProteinCartography](https://github.com/Arcadia-Science/ProteinCartography/)
+- Python packages like [plm-utils](https://github.com/Arcadia-Science/2024-plm-utils).
+- Tools designed to analyze experimental data in high throughput like [arcadia-phenotypeomat-protocol]([https://github.com/Arcadia-Science/arcadia-phenotypeomat-protocol)
+
+## Things to look for when deciding to include a tool as a dependency
+
+A dependency is any software tool that is required to make your tool work. For example, BLAST is a dependency of ProteinCartography and OrthoFinder is a dependency of NovelTree. Below we include some signals for assessing whether a software tool is high quality.
+
+### Critical — we can’t use the tool if these criteria are not met
+
+- **The tool has a license and the license is permissible for commercial use.** Common licenses like BSD2, BSD3, GPL3, and MIT can all be used. Some other licenses can be used as well.
+  - Hint: you can search for the word "commercial" in the license. This will often take you to text about whether the software can be used commercially. If it can’t, it will usually say something like "for non-commercial use only."
+- **The tool can be run on the command line without human input.** A Graphical User Interface (GUI) (including web applications like Streamlit) is not required to run the tool. The tool isn’t encoded in notebooks.
+- **The tool can be run on a Linux platform.** This is a requirement\* for things dockerization and for running on AWS.
+  - *It’s not actually a*requirement\* — the Mac operating systems can’t really be dockerized, but Windows can. Similarly, there are Windows operating systems on AWS but these are sometimes annoying to use.
+
+### Important — think carefully before proceeding with the tool if these criteria are not met
+
+- **The software code is on GitHub or another popular developer platform like GitLab or BitBucket.** Using these platforms signals that the tool is open source and likely versioned. Sometimes you'll find code on SourceForge. This is ok and potentially workable, but not the best. Sometimes you'll find code in the supplement of a paper. This is not awesome and probably signals that the tool will be hard to install or use but you can still try if you really want to.
```

Comment:
I would make this a bit more concrete by explaining *why* code found only on sourceforge and in supplements is not ideal and how to go about deciding when to use it anyway (e.g. there's a tradeoff between how essential and unique the code is for the project and how difficult it will be to install in an automated way).

Reply:
added, thank you!